eval/src/expr.rs

```diff
@@ -0,0 +1,121 @@
+use binary_op::eval_binary_op;
+use if_expr::eval_if_expr;
+use lambda::{eval_lambda_application, eval_lambda_defenition};
+use list_constructor::eval_list;
+use object_constructor::eval_object;
+use parser::ast::Expression;
+use unary_op::eval_unary_op;
+
+use crate::{
+    context::Context,
+    errors::{EvalError, RuntimeError, TypeError, TypeErrorKind},
+    jml_type::JmlType,
+    value::JmlValue,
+};
+
+pub mod binary_op;
+pub mod if_expr;
+pub mod lambda;
+pub mod list_constructor;
+pub mod object_constructor;
+pub mod unary_op;
+
+pub fn eval_expr<'source>(
+    expression: Expression<'source>,
+    ctx: &mut Context<'source>,
+) -> Result<JmlValue<'source>, EvalError> {
+    let Expression { l, r, node } = expression;
+    let span = (l, r - l);
+    match node {
+        parser::ast::ExpressionKind::Null => Ok(JmlValue::null()),
+        parser::ast::ExpressionKind::Float(v) => Ok(JmlValue::float(v)),
+        parser::ast::ExpressionKind::Bool(v) => Ok(JmlValue::bool(v)),
+        parser::ast::ExpressionKind::Int(v) => Ok(JmlValue::int(v)),
+        parser::ast::ExpressionKind::String(v) => Ok(JmlValue::string(v)),
+        parser::ast::ExpressionKind::Object(data) => eval_object(data, ctx),
+        parser::ast::ExpressionKind::List(elems) => eval_list(elems, ctx),
+        parser::ast::ExpressionKind::Variable(ident) => eval_variable(span, ident, ctx),
+        parser::ast::ExpressionKind::IndexAccess { target, index } => {
```

Comment:
Why not move out index access and selector to separate functions? Like you did with other cases?

Reply:
fixed